bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -521,6 +523,49 @@ impl Timeline {
         Ok(())
     }
 
+    /// Sends a user's location as a beacon based on their last beacon_info event.
+    ///
+    /// Retrieves the last beacon_info from the room state and sends a beacon with the geo_uri.
+    /// Since only one active beacon_info per user per room is allowed, we can grab the currently live
+    /// beacon_info event from the room state.
+    ///
+    /// TODO: Does the logic belong in self.inner.room() or here?
+    pub async fn send_user_location_beacon(
+        self: Arc<Self>,
+        geo_uri: String,
+    ) -> Result<(), ClientError> {
+        let Some(raw_event) = self
+            .inner
+            .room()
+            .get_state_event_static_for_key::<BeaconInfoEventContent, _>(
+                self.inner.room().own_user_id(),
+            )
+            .await?
+        else {
+            todo!("How to handle case of missing beacon event for state key?")
+        };
+
+        match raw_event.deserialize() {
+            Ok(SyncOrStrippedState::Sync(SyncStateEvent::Original(beacon_info))) => {
```

Comment:
I believe the only `SyncState` that I care about is `SyncStateEvent::Original``?